src/main/java/com/liferay/upgrades/analyzer/project/dependency/detector/JSPortletProjectDetector.java

```diff
@@ -0,0 +1,49 @@
+package com.liferay.upgrades.analyzer.project.dependency.detector;
+
+import com.liferay.upgrades.analyzer.project.dependency.graph.builder.ProjectsDependencyGraphBuilder;
+import com.liferay.upgrades.analyzer.project.dependency.model.ProjectKey;
+import com.liferay.upgrades.analyzer.project.dependency.util.AnalyzerUtil;
+
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class JSPortletProjectDetector implements ProjectDetector {
+
+    @Override
+    public boolean matches(String fileName, Path file) {
+        return fileName.equals("package.json") && _validateJsonContent(file);
+    }
+
+    @Override
+    public void detect(Path file, ProjectsDependencyGraphBuilder projectsDependencyGraphBuilder) {
+        projectsDependencyGraphBuilder.addProject(
+                _getProjectKey(file),  Collections.emptySet());
+    }
+
+    private ProjectKey _getProjectKey(Path file) {
+        String path = file.getParent().toUri().getPath();
+        String fileName = file.getParent().getFileName().toString();
+
+        String key = AnalyzerUtil.getProjectKeyFormated(fileName);
+
+        return new ProjectKey(key, path);
+    }
+
+    private boolean _validateJsonContent(Path contentPath) {
+        String jsonContent = AnalyzerUtil.getContentByPath(contentPath);
+
+        if (jsonContent.isEmpty()) {
+            return false;
+        }
+
+        Matcher matcher = _JS_PORTLET_PROJECT_PATTERN.matcher(jsonContent);
```

Comment:
will this pattern match which of this patterns below:

1. 
```
      {"portlet": "blablabla"}
```

2.

```
{
"portlet": "coolPortletName"
}
```
3.

```
{
      "blabla" : [
            {"portlet": "blablabla"}
        ] 
}
```

I think there is a specific schema for the JSPortlet configuration.